exist-core/src/main/java/org/exist/xquery/value/Type.java

```diff
@@ -41,7 +41,7 @@
  */
 public class Type {
 
-    public static final int NODE = -1;
+    public static final int NODE = 7;
```

Comment:
@adamretter  What is the technical reason of changing this value?

Reply:
By keeping this value as a positive value, it means that all Type values can be stored into 16 bits, as they are then effectively uint16_t. It also means that in future for Union Types, we can do simple bit operations on them.

We could actually go one better here, and renumber some of the others so that we can fit all the types into 8 bits, i.e. values in the integer range 1 to 254

Reply:
I see. I think it would make sense to make those changes now as we do a major release anyway. So there such a (I think) breaking change is ok.

So push that commit here too.